### ISSUE TITLE
ICE: LLVM ERROR: Instruction Combining seems stuck in an infinite loop after 1000 iterations.

### ISSUE THREAD

**Issue by Vsevolod-Livinskij** [ispc/ispc#1788]
It looks like ISPC encounters some error in LLVM.
Error:
> \>$ispc -O2 --target=avx2-i64x4 func.ispc
LLVM ERROR: Instruction Combining seems stuck in an infinite loop after 1000 iterations.

Reproducer:
```c++
int8 a[1][3][6];
void b(uniform int32 c, uniform int8 d[][3][6]) {
  foreach (e = 0...c)
    for (int8 f = 0;;)
      for (int g = 0;;)
        a[e][f][g] = min(~d[e][f][g], 0);
}
```

ISPC version:
1.14.0dev (build commit fe1d9f5a3f69346b @ 20200607, LLVM 10.0.0)

**Comment by dbabokin** [comment]
@DeepakRajendrakumaran please pass the reproducer to LLVM folks.

**Comment by DeepakRajendrakumaran** [comment]
https://bugs.llvm.org/show_bug.cgi?id=46271

**Comment by dbabokin** [comment]
The issue was resolved, we should port the fix to 10.0.
